soc/infineon/psoc4/psoc4100tp/Kconfig.soc

```diff
@@ -0,0 +1,279 @@
+# Copyright (c) 2025 Infineon Technologies AG,
+# or an affiliate of Infineon Technologies AG.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# MCU series
+config SOC_SERIES_PSOC4100TP
+	bool
+	select SOC_FAMILY_INFINEON_PSOC4
+	help
+	  PSOC4100tp Series MCU
+
+config SOC_SERIES
+	default "psoc4100tp" if SOC_SERIES_PSOC4100TP
+
+# SOC Packages for Infineon PSOC4100Tp series MCUs
+config SOC_PACKAGE_PSOC4100TP_64_TQFP
```

Comment:
`SOC_PACKAGE_*` Kconfigs to be moved to `Kconfig` along with their selection (the SOC_SERIES_* selection remains in this file)